components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt

```diff
@@ -61,6 +62,13 @@ class EditToolbar(
             updateClearViewVisibility(text)
             editListener?.onTextChanged(text)
         }
+
+        setOnDispatchKeyEventPreImeListener { event ->
+            if (event?.keyCode == KeyEvent.KEYCODE_BACK) {
+                editListener?.onCancelEditing()
```

Comment:
This calls the `editListener` instance directly and `onEditCancelled` in `BrowserToolar` never gets called?

Reply:
Fixed, good catch! My previous test I had written would have caught this, but unfortunately there's no way for us to simulate a `KEYCODE_BACK` press 😕